rfcs/0004-checkpoints.md

```diff
@@ -285,6 +295,42 @@ Usage: slatedb --path <PATH> list-checkpoints
 
 ```
 
+### Creating a Checkpoint
+
+Checkpoints can be created using the `create_checkpoint` API, the `create-checkpoint` CLI command, or by the
+writer/reader clients as part of their normal operations (see below sections).
+
+Checkpoints can be taken form the current manifest, or from an existing checkpoint. Checkpoints cannot be created from
+arbitrary manifest versions.
+
+Generally to take a checkpoint from the current manifest, the client runs a procedure that looks like the following:
+1. Get the current manifest M at version V.
+2. If M.`initialized` is false, exit with error.
+3. Compute a new v4 UUID id for the checkpoint.
+4. Create a new entry in `checkpoints` with:
+    - `id` set to the id computed in the previous step
+    - `manfiest_id` set to V or V+1. We use V+1 to allow addition of the checkpoint to be included with other updates.
+    - other fields set as appropriate (e.g. expiry time based on relevant checkpoint creation params)
+5. Write a new manifest M' at version V+1. If CAS fails, go to step 1.
+
+To take a checkpoint from an existing checkpoint with id S, the client runs the following procedure:
+1. Get the current manifest M at version V.
+2. Look for the checkpoint C with id S under `checkpoints`. If there is no such checkpoint, exit with error.
```

Comment:
Do we need to check initialized? Or no because a checkpoint implies initialized is true?

Reply:
The latter. Since we're gating checkpoints on `initialized` it should always be safe to take a checkpoint from an existing checkpoint.